lib/active_record_doctor/detectors/incorrect_length_validation.rb

```diff
@@ -55,6 +57,20 @@ def maximum_allowed_by_validations(model, column)
         end
         length_validator ? length_validator.options[:maximum] : nil
       end
+
+      def column_limit(table, column)
+        return column.limit if column.limit
```

Comment:
There's an edge case we need to handle: a column can have **both** a limit and a check constraint. I suggest we add a method `check_constraint_length_limit` and pick the minimum of `column.limit` and `check_constraint_length_limit(table, column)`.